Dockerfile

```diff
@@ -0,0 +1,25 @@
+# Build the manager binary
+FROM golang:1.21 as builder
+
+WORKDIR /workspace
+
+COPY go.mod go.mod
+COPY go.sum go.sum
+
+RUN go mod download
+
+COPY cmd/ cmd/
+COPY k8s/ k8s/
+COPY modules/ modules/
+COPY provider/ provider/
+
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -o k8s-provider cmd/cli/main.go
```

Comment:
We'll want to build the server `cmd/server/main.go` rather than the CLI tool here.